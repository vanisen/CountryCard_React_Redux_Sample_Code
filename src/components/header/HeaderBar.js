import React from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const styles = {
  root: {
    flexGrow: 1,
  },
  grow: {
    flexGrow: 1,
  },
  menuButton: {
    marginLeft: -12,
    marginRight: 20,
  },
};

function HeaderBar(props) {
  const { classes } = props;
  return (
    <div className={classes.root}>
      <AppBar position="static">
        <Toolbar>
          <Typography variant="h6" color="inherit" className={classes.grow}>
            Countries of the World
          </Typography>
        </Toolbar>
      </AppBar>
    </div>
  );
}

HeaderBar.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(HeaderBar);