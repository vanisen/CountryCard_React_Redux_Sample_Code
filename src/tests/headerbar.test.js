import React from 'react';
import { shallow, mount } from 'enzyme';
import HeaderBar from '../components/header/HeaderBar';
import Toolbar from '@material-ui/core/Toolbar';
import { AppBar } from '@material-ui/core';
import renderer from 'react-test-renderer';

it('HeaderBar renders without crashing', () => {
    shallow(<HeaderBar />);
});

it('HeaderBar renders AppBar and ToolBar properly', () => {
    const wrapper = mount(<HeaderBar />);
    expect(wrapper.contains(AppBar)).toEqual(true);
    expect(wrapper.contains(Toolbar)).toEqual(true);
  });

  it('HeaderBar renders header title properly', () => {
    const wrapper = mount(<HeaderBar />);
    expect(wrapper.find('h6').length).toEqual(1);
  });


it('HeaderBar snapshot renders correctly', () => {
  const tree = renderer
    .create(<HeaderBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});