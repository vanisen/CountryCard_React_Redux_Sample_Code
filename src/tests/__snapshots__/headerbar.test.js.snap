// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBar snapshot renders correctly 1`] = `
<div
  className="HeaderBar-root-1"
>
  <header
    className="MuiPaper-root-13 MuiPaper-elevation4-19 MuiAppBar-root-4 MuiAppBar-positionStatic-8 MuiAppBar-colorPrimary-11"
  >
    <div
      className="MuiToolbar-root-40 MuiToolbar-regular-42 MuiToolbar-gutters-41"
    >
      <h6
        className="MuiTypography-root-44 MuiTypography-h6-61 MuiTypography-colorInherit-73 HeaderBar-grow-2"
      >
        Countries of the World
      </h6>
    </div>
  </header>
</div>
`;
